src/ui/base-page.js

```diff
@@ -0,0 +1,259 @@
+                options: {
+                    components: {
+                        prefsEditorLoader: {
+                            options: {
+                                components: {
+                                    prefsEditor: {
+                                        options: {
+                                            listeners: {
+                                                "onCreate.escalate": "{page}.events.onUioReady.fire",
+                                                "{messageLoader}.events.onResourcesLoaded": [{
+                                                    func: "{that}.events.onPrefsEditorRefresh",
+                                                    namespace: "rerenderUIO"
+                                                },
+                                                {
+                                                    func: "{page}.events.onUioReady.fire",
+                                                    namespace: "escalate"
+                                                },
+                                                {
+                                                    funcName: "sjrk.storyTelling.base.page.updateUioPanelLanguages",
+                                                    args: ["{prefsEditorLoader}", "{page}"],
+                                                    priority: "before:rerenderUIO",
+                                                    namespace: "updateMessageBases"
+                                                }]
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    });
+
+    sjrk.storyTelling.base.page.renderAllUiTemplates = function (component) {
+        fluid.each(component, function (subcomponent) {
+            if (subcomponent &&
+                subcomponent.options &&
+                subcomponent.options.gradeNames &&
+                subcomponent.options.gradeNames.includes("sjrk.storyTelling.ui")) {
+                subcomponent.templateManager.events.onResourceLoadRequested.fire();
+            }
+        });
+    };
+
+    sjrk.storyTelling.base.page.getStoredPreferences = function (pageComponent, cookieStore) {
+        var promise = cookieStore.get();
+
+        promise.then(function (response) {
+            pageComponent.applier.change("", response);
+            pageComponent.events.onPreferencesLoaded.fire();
+        }, function (error) {
+            pageComponent.events.onPreferenceLoadFailed.fire(error);
+        });
+    };
+
+    sjrk.storyTelling.base.page.reloadUioMessages = function (lang, uioMessageLoaderComponent, uioMessageLoaderLocalePath) {
+        // Set the language in the resource loader
+        fluid.set(uioMessageLoaderComponent, uioMessageLoaderLocalePath, lang);
```

Comment:
This function is the source of the race which is causing the tests to fail. Firstly this line updates the component's top-level options which are notionally immutable - then the loadResources line starts a race against the very old-fashioned framework's localisation code and the existing I/O that is already pending at startup. My best feeling at the moment is that it will be less work to update you to my in-progress branch where all of the resource loading code has been rewritten rather than get to the bottom of the inherently race-prone mutation-fest that we have inside old-style fluid.fetchResources. I've got a few fixes to make first before my branch is a drop-in replacement but there should be something testable by tomorrow.